lib/SILOptimizer/IPO/DeadFunctionElimination.cpp

```diff
@@ -374,6 +374,7 @@ class DeadFunctionElimination : FunctionLivenessComputation {
         return false;
       });
     }
+    Module->recalculateVTableEntryCache();
```

Comment:
Can do that, will create such a version, my rationale was mitigating the risk of leaving the table in an inconsistent state (By removing an entry from cache but keeping it in the vtable) - but, your solution is more efficient compilation-time wise.